### PR TITLE
[FVM] beyond EVM part 6.1 - Implement EVMAddress.deposit

### PR DESCRIPTION
**Pull request by turbolent**
Depends on #4866

### CODE REVIEW

**Review by ramtinms — APPROVED** [review]

File: fvm/evm/stdlib/contract.cdc [hunk 1]
```diff
@@ -9,33 +11,45 @@ contract EVM {
...
+        /// Constructs a new EVM address from the given byte representation
         init(bytes: [UInt8; 20]) {
             self.bytes = bytes
         }
+
+        /// Deposits the given vault into the EVM account with the given address
+        access(all)
+        fun deposit(from: @FlowToken.Vault) {
+            let amount = from.balance
+            destroy from
```
Comment (janezpodhostnik) [inline fvm/evm/stdlib/contract.cdc:23]: I guess this is another thing that makes the total supply field on the FlowToken contract useless.

**Review by janezpodhostnik — APPROVED** [review]

File: fvm/evm/stdlib/contract.cdc (hunk 1 above)
Comment (turbolent) [reply fvm/evm/stdlib/contract.cdc:23]: Not sure I understand, can you elaborate?

Comment (janezpodhostnik) [reply fvm/evm/stdlib/contract.cdc:23]: `destroy from` will update the total supply of FlowTokens, but they are not actually being burned, just transformed into another form. Which means `FlowToken.totalSupply` wont be useful anymore, as it looses its meaning.

File: fvm/evm/stdlib/contract.cdc [hunk 2]
```diff
@@ -9,33 +11,45 @@ contract EVM {
...
-        /// Constructs a new EVM address from the given byte representation.
+        /// Constructs a new EVM address from the given byte representation
         init(bytes: [UInt8; 20]) {
             self.bytes = bytes
         }
+
+        /// Deposits the given vault into the EVM account with the given address
+        access(all)
+        fun deposit(from: @FlowToken.Vault) {
+            let amount = from.balance
```
Comment (bluesign) [inline fvm/evm/stdlib/contract.cdc:22]: I would add `from.balance = 0.0` just incase

File: fvm/evm/stdlib/contract.cdc (hunk 1 above)
Comment (turbolent) [reply fvm/evm/stdlib/contract.cdc:23]: This has been resolved in a follow up PR, #4986, see https://github.com/onflow/flow-go/pull/4986/files#diff-02837edc597afad5bd8f7d1962c129bded25adcc21a9889382c7aaad9a0e5b34L23-R26

File: fvm/evm/stdlib/contract.cdc (hunk 2 above)
Comment (turbolent) [reply fvm/evm/stdlib/contract.cdc:22]: This has been addressed in a follow up PR, #4986, see https://github.com/onflow/flow-go/pull/4986/files#diff-02837edc597afad5bd8f7d1962c129bded25adcc21a9889382c7aaad9a0e5b34L23-R26

**Pull request merged**